src/main/java/org/alfresco/util/ScriptUtil.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * #%L
+ * Alfresco Repository
+ * %%
+ * Copyright (C) 2005 - 2019 Alfresco Software Limited
+ * %%
+ * This file is part of the Alfresco software.
+ * If the software was purchased under a paid Alfresco license, the terms of
+ * the paid license agreement will prevail.  Otherwise, the software is
+ * provided under the following open source license terms:
+ *
+ * Alfresco is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Alfresco is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Alfresco. If not, see <http://www.gnu.org/licenses/>.
+ * #L%
+ */
+package org.alfresco.util;
+
+import java.io.IOException;
+import java.io.LineNumberReader;
+
+import org.springframework.core.io.support.EncodedResource;
+
+public abstract class ScriptUtil
+{
+    private static final String DEFAULT_SCRIPT_COMMENT_PREFIX = "--";
+
+    /**
+     * Read a script from the provided EncodedResource and build a String containing
+     * the lines.
+     *
+     * @param resource
+     *            the resource (potentially associated with a specific encoding) to
+     *            load the SQL script from
+     * @return a String containing the script lines
+     */
+    public static String readScript(EncodedResource resource) throws IOException
+    {
+        return readScript(resource, DEFAULT_SCRIPT_COMMENT_PREFIX);
+    }
+
+    /**
+     * Read a script from the provided EncodedResource, using the supplied line
+     * comment prefix, and build a String containing the lines.
+     *
+     * @param resource
+     *            the resource (potentially associated with a specific encoding) to
+     *            load the SQL script from
+     * @param lineCommentPrefix
+     *            the prefix that identifies comments in the SQL script (typically
+     *            "--")
+     * @return a String containing the script lines
+     */
+    private static String readScript(EncodedResource resource, String lineCommentPrefix) throws IOException
+    {
+        LineNumberReader lineNumberReader = new LineNumberReader(resource.getReader());
```

Comment:
Looking at https://github.com/Alfresco/alfresco-repository/blob/5e42f708be424de28864051917c53a48c8eada6c/src/main/java/org/alfresco/repo/domain/schema/SchemaBootstrap.java#L1268 we might need to care about UTF-8 encoding.

Reply:
Yes, I updated retrieveAndProcessSequences to use UTF-8:
`final String script = DBScriptUtil.readScript(new EncodedResource(resource, "UTF-8"));`